demo/controls/InputDemo.jsx

```diff
@@ -21,6 +24,10 @@ var dropDownData = [
 ];
 
 var InputDemo = React.createClass({
+  textFieldSearch: null,
+  onFocusTextSearchBox: function () {
```

Comment:
I think the onFocus feature should be inside the TextField component. TextField component should accept a prop where it will determine if the field should be auto focused or not.